include/wasm.h

```diff
@@ -9,6 +9,13 @@
 #include <string.h>
 #include <assert.h>
 
+#ifndef WASM_C_API
```

Comment:
Could we call it something less generic, e.g. `WASM_C_API_EXPORT`?

Reply:
...or `WASM_API_EXPORT` to match recent changes (i.e. `WASM_API_DEBUG`).

Reply:
I think `WASM_API_EXPORT` would be strange in the case where it's defined `WASM_API_EXPORT=__declspec(dllimport)`.

Perhaps `WASM_API_LINKAGE`?

Reply:
Fair enough, although [a lot of existing projects use `#define xxx_EXPORT __declspec(dllimport)`](https://github.com/search?q=%22export+__declspec%28dllimport%29%22&type=Code).

`#define xxx_EXTERN __declspec(dllimport)` also seems to be a popular choice.

@rossberg any preference?

Reply:
@rossberg ping

Reply:
Can't say I'm thrilled about the pervasive pollution with Windows nonsense, but if necessary, I'd go with WASM_API_EXTERN.

Reply:
> Windows nonsense

Having the annotation is also required for building shared libraries with `-fvisibility=hidden` on Unix-like systems; it's just that for those, the embedder's definition of the macro is empty (so it just takes the #else branch here), while the library's definition will be `__attribute__ ((visibility("default")))`.